cmd/ttn-lw-cli/commands/end_devices.go

```diff
@@ -466,17 +480,13 @@ var (
 				return nil
 			}
 			var device ttnpb.EndDevice
-			if ttnpb.HasAnyField(ttnpb.TopLevelFields(paths), "root_keys") {
-				device.SupportsJoin = true
-				paths = append(paths, "supports_join")
-			}
 			if err = util.SetFields(&device, setEndDeviceFlags); err != nil {
 				return err
 			}
 			device.Attributes = mergeAttributes(device.Attributes, cmd.Flags())
 			device.EndDeviceIdentifiers = *devID
 
-			isPaths, nsPaths, asPaths, jsPaths := splitEndDeviceSetPaths(device.SupportsJoin, paths...)
+			isPaths, nsPaths, asPaths, jsPaths := splitEndDeviceSetPaths(ttnpb.HasAnyField(paths, setEndDeviceToJS...), paths...)
 
 			if len(nsPaths) > 0 {
```

Comment:
Here, and below we should check for `config.NetworkServerEnabled` etc.

Reply:
Where exactly?

Which checks below?

Because `setEndDevice()` does this already and prints warnings if fields are being updated while the concerning components aren't enabled.

Reply:
With `if len(nsPaths) > 0 {`, `if len(asPaths) > 0 {` and `if len(jsPaths) > 0 {`.
We don't need to ask the IS for the NS/AS/JS address if we're not going to use it.

Reply:
https://github.com/TheThingsNetwork/lorawan-stack/pull/844/commits/da60c91bea5e9e99353039715ac33e6a51f5dd5c